src/schemathesis/runner/checks.py

```diff
@@ -1,31 +1,35 @@
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Union
 
 import jsonschema
 import requests
 
 from ..models import TestResult
-from ..utils import are_content_types_equal
+from ..utils import WSGIResponse, are_content_types_equal
 
+GenericResponse = Union[requests.Response, WSGIResponse]
 
-def not_a_server_error(response: requests.Response, result: TestResult) -> None:
+
+def not_a_server_error(response: GenericResponse, result: TestResult) -> None:
     """A check to verify that the response is not a server-side error."""
-    assert response.status_code < 500, f"Received a response with 5xx status code: {response.status_code}"
+    if response.status_code >= 500:
+        raise AssertionError(f"Received a response with 5xx status code: {response.status_code}")
```

Comment:
This one could be done separately - quite strange behavior. Asserts here were rewritten by pytest for some reason